blockly-core/core/code_generation/generator.js

```diff
@@ -197,6 +197,11 @@ Blockly.CodeGenerator.prototype.blockToCode = function(block, opt_showHidden) {
   if (!block) {
     return '';
   }
+  if (block.skipGeneration) {
+    // Clear the flag, so it doesn't persist to future serializations.
+    block.skipGeneration = false;
```

Comment:
it makes me a little uncomfortable that `blockToCode` now has such a side effect, particularly given how commonly used it is. Is there a better way to do this?